### ISSUE TITLE
Style/EmptyLiteral + frozen string literals

### ISSUE THREAD

**Issue by chriseppstein** [bbatsov/rubocop#2645]
What is the recommendation for how to handle frozen string literals because the EmptyLiteral cop fails on String.new but there's no way to disable just the check for empty string literals.

**Comment by jonas054** [comment]
It seems to me that auto-correcting `String.new` into `""` can produce erroneous code, if you're running with frozen string literal and it was a non-frozen String you wanted. Those bugs might be hard to track down too. I think we need to remove that auto-correction, and maybe the entire offense since we can't know if it's safe to change `String.new` into `""`.

**Comment by segiddins** [comment]
We _can_ know if it's safe to change `String.new` into `""` though -- if frozen string literals are enabled. And there's a FrozenStringLiteral mixin that can tell us just that

**Comment by alexdowad** [comment]
Don't understand what this issue is about.

**Comment by jonas054** [comment]
@alexdowad Let's say you have this file:
```ruby
# encoding: utf-8
# frozen_string_literal: true

s = String.new
s << "Hello"
puts s
```
`rubocop -a` will change `s = String.new` into `s = ''` and the program no longer works with ruby >= 2.3.

@segiddins's solution sounds good to me.

**Comment by alexdowad** [comment]
Thanks. But when @segiddins said "it's safe if frozen string literals are enabled", did he actually mean "if frozen string literals are not enabled"?

**Comment by segiddins** [comment]
I think I did
